channeldb/payment_control.go

```diff
@@ -188,6 +200,14 @@ func (p *PaymentControl) RegisterAttempt(paymentHash lntypes.Hash,
 		var check []inflightCheck
 		check = append(check, ensureNonTerminal())
 
+		// We must also check that registering this attempt doesn't
+		// take the total sent amount above the payment value.
+		if info.Value < attempt.Route.ReceiverAmt() {
+			return ErrValueExceedsAmt
+		}
+		maxSent := info.Value - attempt.Route.ReceiverAmt()
+		check = append(check, ensureMaxSent(maxSent))
```

Comment:
Instead of those functional options, can't you just fetch the payment and then execute the checks in the old-fashioned way?

Reply:
done.